src/test/java/ch/jalu/collectionbehavior/ListTest.java

```diff
@@ -240,103 +323,62 @@ void jdkCollectionsUnmodifiableList() {
         assertThat(list, instanceOf(RandomAccess.class)); // Because wrapped list is RandomAccess, too.
         assertThat(new LinkedList<>(), not(instanceOf(RandomAccess.class))); // validate assumption
         assertThat(Collections.unmodifiableList(new LinkedList<>()), not(instanceOf(RandomAccess.class)));
-
-        // May contain null
-        List<String> listWithNull = Arrays.asList("a", null, "c");
-        assertThat(Collections.unmodifiableList(listWithNull), contains("a", null, "c"));
-
-        // Null support in methods
-        verifySupportsNullArgInMethods(list);
-
-        // Same instance is returned in JDK  17, whereas in JDK 11 it always created a new instance
-        assertThat(Collections.unmodifiableList(list), sameInstance(list));
     }
 
     /**
      * {@link Collections#singletonList} provides a list with a single given element. Immutable. Supports null.
      */
-    @Test
-    void jdkCollectionsSingletonList() {
-        // Is immutable
-        List<String> list = Collections.singletonList("a");
-        verifyThrowsOnlyIfListWouldBeModified(list, UnmodifiableListExceptionBehavior.COLLECTIONS_SINGLETONLIST);
-
-        // Implements RandomAccess
-        assertThat(list, instanceOf(RandomAccess.class));
-
-        // May contain null
-        assertThat(Collections.singletonList(null), contains((Object) null));
-
-        // Null support in methods
-        verifySupportsNullArgInMethods(list);
+    @TestFactory
+    List<DynamicTest> jdk_Collections_singletonList(TestInfo testInfo) {
+        return TestsGenerator.forType(testInfo, ListCreator.forSingleElement(Collections::singletonList))
+            .expect(NullSupport.FULL, true)
+            .mutability(ListModificationBehavior.immutable().throwsIfWouldBeModified(false)
```

Comment:
`throwsIfWouldBeModified(false)` is misleading to read -> find other way to define if IIO is thrown